File: reference-benchmark/sequila/sequila-job.py
import os

from pysequila import SequilaSession

ss = (SequilaSession
      .builder
      .config("spark.jars.packages", "org.biodatageeks:sequila_2.12:1.1.0")
      .config("spark.driver.memory", "4g")
      .getOrCreate()
      )

database_path = os.getenv("SQ_DATA_PATH_DB")
query_path = os.getenv("SQ_DATA_PATH_Q")
out_path = os.getenv("SQL_DATA_OUT")

database = ss.read.format("org.biodatageeks.sequila.datasources.BED.BEDDataSource").load(database_path)
query = ss.read.format("org.biodatageeks.sequila.datasources.BED.BEDDataSource").load(query_path)

overlap_cond = (database.pos_end >= query.pos_start) & (database.pos_start <= query.pos_end)
result = (database.join(query, (database.contig == query.contig) & overlap_cond))
result.write.csv("out_path", mode="append")

input("Press Enter to continue...")

File: reference-benchmark/sequila/sequila-job.scala
import org.apache.spark.sql
import org.apache.spark.sql.DataFrame
import org.apache.spark.sql.SparkSession
import org.apache.spark.sql.SequilaSession

import org.apache.spark.sql.{functions => F}

val path = "/Users/iw-user/Git/FEIT/Interwalled/interwalled-data-generator/data/maximum-size-of-input"

val spark      = SparkSession.builder().appName("Interval benchmark").getOrCreate()
val sqlsession = SequilaSession(spark)

def parse(input: DataFrame): DataFrame =
  input
    .withColumn("text", F.split(F.col("value"), "\\s+"))
    .select(
      F.col("text").getItem(0).alias("contig"),
      F.col("text").getItem(1).alias("pos_start"),
      F.col("text").getItem(2).alias("pos_end"),
    )

val database   = sqlsession.read
  .text(path + "/input-80M.bed")
  .transform(parse)
  .alias("database")

val query      = sqlsession.read
  .text(path + "/input-80M.bed")
  .transform(parse)
  .alias("query")


database
  .join(query,
    (database.col("pos_start") <= query.col("pos_end")) &&
    (database.col("pos_end") >= query.col("pos_start")) &&
    (database.col("contig") === query.col("contig"))
  )
  .select(
    database.col("pos_start").alias("lhs_start"),
    database.col("pos_end").alias("lhs_end"),
    query.col("pos_start").alias("rhs_start"),
    query.col("pos_end").alias("rhs_end"),
    database.col("contig")
  )
  .foreach (_ => ())

scala.io.StdIn.readLine()